packages/block-library/src/image/edit.js

```diff
@@ -363,6 +356,14 @@ export function ImageEdit( {
 		ref: containerRef,
 		className: classes,
 	} );
+	// Only observe the max width from the parent container when the parent layout is not flex nor grid.
+	// This won't work for them because the container width changes with the image.
+	// TODO: Find a way to observe the container width for flex and grid layouts.
+	const isMaxWidthContainerWidth =
+		! parentLayout || ! SIZED_LAYOUTS.includes( parentLayout.type );
+	const [ maxWidthObserver, maxContentWidth ] = useMaxWidthObserver(
+		blockProps.className
+	);
```

Comment:
This is just moved down so the `blockProps` are available for passing in the `className`